.devcontainer/devcontainer.json

```diff
@@ -33,20 +49,42 @@
             ]
         }
     },
-    "features": {
-        // add in eternal history and other bash features
-        "ghcr.io/diamondlightsource/devcontainer-features/bash-config:1": {}
-    },
-    // Create the config folder for the bash-config feature
-    "initializeCommand": "mkdir -p ${localEnv:HOME}/.config/bash-config",
+    // Create the config folder for the bash-config feature and uv cache
+    "initializeCommand": "mkdir -p ${localEnv:HOME}/.config/terminal-config",
     "runArgs": [
         // Allow the container to access the host X11 display and EPICS CA
         "--net=host",
         // Make sure SELinux does not disable with access to host filesystems like tmp
         "--security-opt=label=disable"
     ],
+    "mounts": [
+        // Mount in the user terminal config folder so it can be edited
+        {
+            "source": "${localEnv:HOME}/.config/terminal-config",
+            "target": "/user-terminal-config",
+            "type": "bind"
+        },
+        // Keep a persistent cross container cache for uv, pre-commit, and the venvs
+        {
+            "source": "persistent-uv-cache",
+            "target": "/root/.cache/uv",
+            "type": "volume"
+        },
+        // ...and for pre-commit
+        {
+            "source": "persistent-pre-commit-cache",
+            "target": "/root/.cache/pre-commit",
+            "type": "volume"
+        }
+        // Use a volume mount for the uv venv so it is local to the container
+        {
+            "target": "/workspaces/${localWorkspaceFolderBasename}/.venv",
+            "source": "devcontainer-shared-cache",
+            "type": "volume"
+        }
```

Comment:
This has been changed upstream to use a single cache for everything, do we want to keep in sync with that?